rswag-specs/spec/rswag/specs/request_factory_spec.rb

```diff
@@ -103,6 +103,50 @@ module Specs
           end
         end
 
+        context "'query' parameters of type 'array' by way of 'schema'" do
+          before do
+            metadata[:operation][:parameters] = [
+              { name: 'things', in: :query, schema: { type: :array }, collectionFormat: collection_format }
+            ]
+            allow(example).to receive(:things).and_return(['foo', 'bar'])
```

Comment:
This is only for OpenAPI 3. This library, afaik, intends to support OpenAPI 2 and 3 side by side. You can see other logic in the request_factory that checks the swagger_doc version in an `if/else`.

```suggestion
            allow(example).to receive(:things).and_return(['foo', 'bar'])
            let(:swagger_doc) { { swagger: '3.0.3' } }
```

Reply:
Thanks for giving this a look, @akofink! 

Based on what I was seeing elsewhere in the `RequestFactory`, I've [force-pushed](https://github.com/rswag/rswag/compare/2aaeda2ec4ea81888de4f37c2af5edb8c04912ec..d463c88049d637f3822a782e885f616e92bd7d76) (and then [force-pushed a correction](https://github.com/rswag/rswag/compare/d463c88049d637f3822a782e885f616e92bd7d76..f614f38f0cc46403f8e9c7b38f156805bfa8a692) 🤦) an update to leave the 2.0 version untouched, but to have OpenAPI 3 be able to use this new format.  

Is that closer to what you were thinking?

Reply:
Yep! Much better, thanks!